modules/core/navigation/navigation.js

```diff
@@ -53,25 +53,42 @@ async function navigate(url, loaderMessage) {
     try {
         const response = await fetch(url, {
             method: 'GET',
+            credentials: 'include',
         });
 
         if (!response.ok) {
             throw new Error("Request failed with status: " + response.status);
         }
 
         const html = await response.text();
-        const main = html.match(/<main[^>]*>((.|[\n\r])*)<\/main>/i)[0];
-        const title = html.match(/<title[^>]*>((.|[\n\r])*)<\/title>/i)[0];
+        if (!html || typeof html !== 'string') {
+            throw new Error("Empty or invalid response body.");
+        }
+
+        const parser = new DOMParser();
+        const doc = parser.parseFromString(html, 'text/html');
+
+        const main = doc.querySelector('main');
+        const title = doc.querySelector('title');
 
+        if (!main || !title) {
+            throw new Error("main or title not found in parsed HTML");
```

Comment:
This error is sent to the end user but I assume that very little number of Cypht end users will actually know what does this mean. Can you improve the error messages here connected with navigation? Maybe add a nice user-facing error msg like 'Failed navigating to page ...' and then output the actual error in the console as a warning or error, so developers can look into that?